managed/services/preconditions_test.go

```diff
@@ -0,0 +1,146 @@
+		err := db.InTransactionContext(context.Background(), &sql.TxOptions{Isolation: sql.LevelSerializable}, func(tx *reform.TX) error {
+			return CheckMongoDBBackupPreconditions(db.Querier, models.Snapshot, "", "service1", "")
+		})
+		tests.AssertGRPCError(t, status.New(codes.FailedPrecondition, "A snapshot backup for service 'service1' can be performed only if there are no other scheduled backups for this service."), err)
+	})
+
+	t.Run("able to update existing PITR backup for service", func(t *testing.T) {
+		err := db.InTransactionContext(context.Background(), &sql.TxOptions{Isolation: sql.LevelSerializable}, func(tx *reform.TX) error {
+			return CheckMongoDBBackupPreconditions(db.Querier, models.PITR, "", "service1", schedule1.ID)
+		})
+		require.NoError(t, err)
+	})
+
+	t.Run("unable to create PITR backup for cluster with scheduled snapshot backup", func(t *testing.T) {
+		err := db.InTransactionContext(context.Background(), &sql.TxOptions{Isolation: sql.LevelSerializable}, func(tx *reform.TX) error {
+			return CheckMongoDBBackupPreconditions(db.Querier, models.PITR, "cluster2", "", "")
+		})
+		tests.AssertGRPCError(t, status.New(codes.FailedPrecondition, "A PITR backup for the cluster 'cluster2' can be enabled only if there are no other scheduled backups for this cluster."), err)
+	})
+
+	t.Run("able to create second snapshot backup for cluster", func(t *testing.T) {
+		err := db.InTransactionContext(context.Background(), &sql.TxOptions{Isolation: sql.LevelSerializable}, func(tx *reform.TX) error {
+			return CheckMongoDBBackupPreconditions(db.Querier, models.Snapshot, "cluster2", "", "")
+		})
+		require.NoError(t, err)
+	})
+
+	t.Run("unable to create PITR backup for service with scheduled snapshot backup", func(t *testing.T) {
+		err := db.InTransactionContext(context.Background(), &sql.TxOptions{Isolation: sql.LevelSerializable}, func(tx *reform.TX) error {
+			return CheckMongoDBBackupPreconditions(db.Querier, models.PITR, "", "service2", "")
+		})
+		tests.AssertGRPCError(t, status.New(codes.FailedPrecondition, "A PITR backup for the service with ID 'service2' can be enabled only if there are no other scheduled backups for this service."), err)
+	})
+
+	t.Run("able to create second snapshot backup for service", func(t *testing.T) {
+		err := db.InTransactionContext(context.Background(), &sql.TxOptions{Isolation: sql.LevelSerializable}, func(tx *reform.TX) error {
+			return CheckMongoDBBackupPreconditions(db.Querier, models.Snapshot, "", "service2", "")
+		})
+		require.NoError(t, err)
+	})
+
+	t.Run("incremental backups are not supported", func(t *testing.T) {
+		err := db.InTransactionContext(context.Background(), &sql.TxOptions{Isolation: sql.LevelSerializable}, func(tx *reform.TX) error {
+			return CheckMongoDBBackupPreconditions(db.Querier, models.Incremental, "cluster1", "", "")
+		})
+		tests.AssertGRPCError(t, status.New(codes.InvalidArgument, "Incremental backups unsupported for MongoDB"), err)
+	})
+}
```

Comment:
just a thought - we have a bunch of tests following the same pattern here, maybe we can convert them to table tests?